### ISSUE TITLE
`gns3-server`: Could not find a version that satisfies the requirement async-timeout==3.0.1

### ISSUE THREAD

**Issue by little-dude** [NixOS/nixpkgs#150269]
### Describe the bug

On nixos-unstable, the `gns3-server` package does not install:

### Steps To Reproduce

```
nix-env -iA nixos.gns3-server
```

### Expected behavior

`gns3-server` is installed

### Notify maintainers

No maintainer but pinging @fabaff  who last touched `gns3/server.nix`

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯  nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/4i653qmy6rs7r3m6zh60j0nwnnc5gi6k-bash-interactive-5.1-p12-dev
copying path '/nix/store/4i653qmy6rs7r3m6zh60j0nwnnc5gi6k-bash-interactive-5.1-p12-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.4, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixos-22.05pre335501.c71f061c68b"`
 - channels(little-dude): `"home-manager, nixos-22.05pre337103.581d2d6c9cd"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by fabaff** [comment]
`async-timeout` was updated to 4.0.1 a while back. `gns3` is pinning most of it's dependencies thus the issue.

Looks like that `gns3` could need an update (available 2.2.148, upstream 2.2.27).

**Comment by little-dude** [comment]
So I can build the package with the following:
```
  postPatch = ''
    substituteInPlace requirements.txt \
      --replace "aiohttp==3.7.4.post0" "aiohttp>=3.7.4" \
      --replace "Jinja2==3.0.1" "Jinja2>=3.0.1" \
      --replace "sentry-sdk==1.3.1" "sentry-sdk>=1.4" \
      --replace "async-timeout==3.0.1" "async-timeout>=4.0" 
  '';
```
But it doesn't look right. I guess we shouldn't blindly replace pinned versions with incompatible versions. How does nixpkgs usually deals with this?
